pipeline/run_models.py

```diff
@@ -7,65 +7,69 @@
     random_forest_regressor_gridsearch,
     xgboost_classifier_gridsearch,
     logistic_regression_gridsearch,
-    statistical_model,
 )
 from pipeline.models.utils.model_enum import Model
 import pipeline.models.utils.scoring as scoring
 import pandas as pd  # type: ignore
 from pipeline.preprocessing.sk_formatter import SKFormatter
-import os
 
 Params = dict[str, Any]
 Models = dict[Model, Params]
+Job = tuple[Model, Callable[[pd.DataFrame, pd.DataFrame], tuple[Any, dict]]]
 
 
-def train_models_save_results(
-    x_train: np.ndarray, y_train: np.ndarray
-) -> dict[Model, Any]:
+def runner(model_jobs: list[Job], formatter: SKFormatter) -> None:
     """
-    Creates every model from models.py, fits them, saves
-    them to pickle, saves best params and returns dict
-    mapping model names to the fitted model.
+    The runner, at a high-level, is responsible for:
+      1. Training the individual models of the model_jobs
+      2. Save the SKFormatter params along side the models themselves, their params and metrics
 
     use joblib for saving models to file:
     # https://scikit-learn.org/stable/model_persistence.html
 
     Args:
-        x_train (np.ndarray): Training dataset
-        y_train (np.ndarray): Target for training
-
-    Returns:
-        dict[Model, Any]: dictionary of model name to trained model
+        model_jobs (list[Job]):
+        formatter (SKFormatter):
     """
+    prefix = datetime.today().strftime("%m%d_%H%M_")
 
-    # define a list of models and their corresponding grid search functions (from models.py)
-    model_jobs: list[
-        tuple[Model, Callable[[pd.DataFrame, pd.DataFrame], tuple[Any, dict]]]
-    ] = [
-        (Model.MLP, create_mlp_grid_search),  # type: ignore
-        (Model.RF, random_forest_regressor_gridsearch),
-        (Model.XGB, xgboost_classifier_gridsearch),
-        (Model.LOGREG, logistic_regression_gridsearch),
-        # (Model.STATMODEL, statistical_model), # TODO: Does not work currently...
-    ]
+    x_train, x_test, y_train, y_test = formatter.generate_train_test_split()
+
+    save_skformatter_params(formatter.params, prefix)
+
+    file = f"/share-files/runs/{prefix}/{prefix}metrics"
+    with open(file, "a") as f:
```

Comment:
```suggestion
    with open(file, "a+") as f:
```

Python skal have permissions til at lave filen hvis den ikke eksistere 🤓